custom-routes-vis/src/main/webapp/script.js

```diff
@@ -246,36 +207,61 @@ function generateRoutes() {
       '&endpoint=' + serviceEndpoint + '&apiKey=' + apiKey + '&rateCard=' +
       JSON.stringify(createRateCard()))
       .then(response => response.json()).then(directions => {
-        // Log the response status from the Directions API.
-        // TODO(chenyuz): Find equivalence for Routes Preferred API.
-        console.log(directions.status);
-
         let routes = directions.routes;
         console.log('num routes:', routes.length);
 
         for (let routeNum = 0; routeNum < routes.length; routeNum++) {
-          createRoutePolyline(routeNum, routes[routeNum], 
-              serviceEndpoint === 'directions');
+          createRoutePolyline(routeNum, routes[routeNum], serviceEndpoint);
         }
       });
 }
 
+/**
+ * Creates formatted rate card object based on form input.
+ * @returns {Object} Rate Card JSON Object.
+ */
+function createRateCard() {
+  let costPerMin = parseFloat(document.getElementById('cost-per-minute').value);
+  let costPerKm = parseFloat(document.getElementById('cost-per-km').value);
+  let includeTolls = ('true' === document.getElementById('include-tolls').value);
+
+  let rateCard = {'include_tolls': includeTolls};
+  if (!isNaN(costPerMin)) {
+    rateCard['cost_per_minute'] = {'value': costPerMin};
+  }
+  if (!isNaN(costPerKm)) {
+    rateCard['cost_per_km'] = {'value': costPerKm};
+  }
+
+  return rateCard;
+}
+
 /**
  * Creates a polyline for a route on the map.
  * @param {num} routeNum The index of the route in the order that it is returned
  * from the selected API.
- * @param {!Object} routeJson JSON object containing all information of the target
- * route.
- * @param {boolean} isDirectionsApi Whether the route is obtained by the Directions
- * API; if false, assume it is obtained by the Routes Preferred API.
+ * @param {!Object} routeJson JSON object containing all information of the 
+ * target route.
+ * @param {boolean} serviceEndpoint The service endpoint selected by the user. 
+ * Can be one of 'directions', 'compute-routes', or 'compute-routes-alpha'.
```

Comment:
```suggestion
 * @param {!Object} routeJson JSON object containing all information of the 
 *     target route.
 * @param {boolean} serviceEndpoint The service endpoint selected by the user. 
 *     Can be either 'directions', 'compute-routes', or 'compute-routes-alpha'.
```
Also fix this in line 242.

Reply:
Fixed the indent, but I thought "either" refers to one of two things but we have three here

Reply:
Okay, let's just delete either than. Just "Can be 'directions'....." etc.

Reply:
sound good